### ISSUE TITLE
iris example in doc results in an error 

### ISSUE THREAD

**Issue by sfd99** [xiaodaigh/disk.frame#241]
_Hi,_

using all latest versions of:
- **disk.frame** pkg version 0.3.0
with **R** 3.6.1, **Rstudio**  1.2.5019 
and **Ubuntu LINUX 18.04** 64 bits.

The IRIS example in your web page,
results in an _error message_ from R:

> result_from_disk.frame = iris %>% 
     as.disk.frame %>% 
     group_by(Species) %>% 
     summarize(
         mean(Petal.Length), 
         sumx = sum(Petal.Length/Sepal.Width), 
         sd(Sepal.Width/ Petal.Length), 
         var(Sepal.Width/ Sepal.Width), 
         l = length(Sepal.Width/ Sepal.Width + 2),
         max(Sepal.Width), 
         min(Sepal.Width), 
         median(Sepal.Width)
     ) %>% 
     collect

  **Error: $ operator is invalid for atomic vectors** 
>

**Comment by xiaodaigh** [comment]
@sfd99 Thank you for the report.

I can't replicate your issue on Windows nor WSL Ubuntu. 

Can you run `devtools::sessionInfo()` and paste the results here?

For example the below works.
```r
library(disk.frame)
# setup_disk.frame()
result_from_disk.frame = iris %>%
  as.disk.frame %>%
  group_by(Species) %>%
  summarize(
    mean(Petal.Length),
    sumx = sum(Petal.Length/Sepal.Width),
    sd(Sepal.Width/ Petal.Length),
    var(Sepal.Width/ Sepal.Width),
    l = length(Sepal.Width/ Sepal.Width + 2),
    max(Sepal.Width),
    min(Sepal.Width),
    median(Sepal.Width)
  ) %>%
  collect
```

**Comment by sfd99** [comment]
Hi 
   - thanks for replying so quickly.

Here's the requested info,
run devtools::sessionInfo():
> sessionInfo()
R version 3.6.2 (2019-12-12)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so

Random number generation:
 RNG:     Mersenne-Twister 
 Normal:  Inversion 
 Sample:  Rounding 
 
locale:
 [1] LC_CTYPE=en_US.UTF-8      
 [2] LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8       
 [4] LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8   
 [6] LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8      
 [8] LC_NAME=C                 
 [9] LC_ADDRESS=C              
[10] LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8
[12] LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices
[4] utils     datasets  methods  
[7] base     

other attached packages:
[1] disk.frame_0.3.0  purrr_0.3.3      
[3] tidyr_1.0.0       wrapr_1.9.3      
[5] DescTools_0.99.31 skimr_2.0.2      
[7] visdat_0.5.3      ggplot2_3.2.1    
[9] dplyr_0.8.3      

loaded via a namespace (and not attached):
 [1] pkgload_1.0.2      bit64_0.9-7       
 [3] jsonlite_1.6       assertthat_0.2.1  
 [5] expm_0.999-4       remotes_2.1.0     
 [7] sessioninfo_1.1.1  globals_0.12.5    
 [9] pillar_1.4.2       backports_1.1.5   
[11] lattice_0.20-38    glue_1.3.1        
[13] digest_0.6.23      pryr_0.1.4        
[15] colorspace_1.4-1   htmltools_0.4.0   
[17] Matrix_1.2-18      pkgconfig_2.0.3   
[19] devtools_2.2.1     listenv_0.8.0     
[21] mvtnorm_1.0-11     scales_1.1.0      
[23] processx_3.4.1     tibble_2.1.3      
[25] usethis_1.5.1      ellipsis_0.3.0    
[27] withr_2.1.2        furrr_0.1.0       
[29] repr_1.0.2         lazyeval_0.2.2    
[31] fst_0.9.0          cli_2.0.0         
[33] magrittr_1.5       crayon_1.3.4      
[35] memoise_1.1.0      ps_1.3.0          
[37] bigassertr_0.1.1   fs_1.3.1          
[39] future_1.15.1      fansi_0.4.0       
[41] MASS_7.3-51.4      pkgbuild_1.0.6    
[43] tools_3.6.2        data.table_1.12.8 
[45] prettyunits_1.0.2  lifecycle_0.1.0   
[47] stringr_1.4.0      munsell_0.5.0     
[49] callr_3.4.0        packrat_0.5.0     
[51] compiler_3.6.2     rlang_0.4.2       
[53] grid_3.6.2         rstudioapi_0.10   
[55] base64enc_0.1-3    boot_1.3-23       
[57] testthat_2.3.1     gtable_0.3.0      
[59] codetools_0.2-16   R6_2.4.1          
[61] knitr_1.26         future.apply_1.3.0
[63] bit_1.1-14         zeallot_0.1.0     
[65] rprojroot_1.3-2    bigreadr_0.2.0    
[67] desc_1.2.0         stringi_1.4.3     
[69] parallel_3.6.2     Rcpp_1.0.3        
[71] vctrs_0.2.1        tidyselect_0.2.5  
[73] xfun_0.11  

I am really eager 
to make your example work! :-)
------------------
**sfd99
San Francisco**

**Comment by xiaodaigh** [comment]
Thanks @sfd99 for the report! It is indeed a bug! When the number of chunks is 1. This has been addressed in #243 

You can verify the below

```r
nchunks( iris %>% as.disk.frame) == 1
```

This only happens if you happen to run `{disk.frame}` on a one-core machine! Which was never test, but have now been added to the test suite!

```r
remotes::install_github("xiaodaigh/disk.frame")
```
and test again should work now!

**Comment by sfd99** [comment]
OK - 
After re-install to latest  (fixed) version:
- Works perfect now! :-)

great job - thanks!

**sfd99
San Francisco**
